File: scripts/lib/resolve-install-manifest.mjs
#!/usr/bin/env node
/**
 * Node resolver for installation profile (ci-parity / node:20-slim — no python3).
 * Keep in sync with scripts/lib/resolve-install-manifest.py
 */
import { readFileSync, existsSync } from "node:fs";
import { resolve, dirname } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));

function repoRoot(start = process.cwd()) {
  let cur = resolve(start);
  for (;;) {
    if (existsSync(resolve(cur, ".umbraculum/install.core.json"))) return cur;
    const parent = resolve(cur, "..");
    if (parent === cur) return start;
    cur = parent;
  }
}

function manifestPath(root, env) {
  const override = env.UMBRACULUM_INSTALL_MANIFEST?.trim();
  if (override) return resolve(override);

  const active = resolve(root, ".umbraculum/install.json");
  if (existsSync(active)) return active;

  const profile = env.UMBRACULUM_MODULE_PROFILE?.trim() || "platform";
  const name = profile === "reference" ? "install.reference.json" : "install.core.json";
  return resolve(root, ".umbraculum", name);
}

function loadManifest(root, env) {
  const path = manifestPath(root, env);
  if (!existsSync(path)) {
    const profile = env.UMBRACULUM_MODULE_PROFILE?.trim() || "platform";
    if (profile === "reference") {
      return { id: "reference", verticals: ["brewery"], nativeApps: ["brewery"] };
    }
    return { id: "core", verticals: [], nativeApps: ["blank"] };
  }
  return JSON.parse(readFileSync(path, "utf8"));
}

const args = process.argv.slice(2);
const fieldIdx = args.indexOf("--field");
const field = fieldIdx >= 0 ? args[fieldIdx + 1] : null;
const rootArgIdx = args.indexOf("--repo-root");
const root = rootArgIdx >= 0 ? resolve(args[rootArgIdx + 1]) : repoRoot();

const manifest = loadManifest(root, process.env);

if (args.includes("--json")) {
  console.log(JSON.stringify(manifest, null, 2));
  process.exit(0);
}

switch (field) {
  case "id":
    console.log(manifest.id ?? "core");
    break;
  case "verticals":
    console.log((manifest.verticals ?? []).join(","));
    break;
  case "nativeApps":
    console.log((manifest.nativeApps ?? []).join(","));
    break;
  case "primaryNativeApp":
    console.log((manifest.nativeApps ?? ["blank"])[0] ?? "blank");
    break;
  case "hasBrewery":
    console.log((manifest.verticals ?? []).includes("brewery") ? "true" : "false");
    break;
  default:
    console.log(JSON.stringify(manifest));
}

File: scripts/lib/resolve-install-manifest.py
#!/usr/bin/env python3
"""Resolve installation profile manifest for shell scripts and verify-slice."""
from __future__ import annotations

import argparse
import json
import os
import sys
from pathlib import Path


def repo_root(start: Path | None = None) -> Path:
    cur = (start or Path.cwd()).resolve()
    for candidate in [cur, *cur.parents]:
        if (candidate / ".umbraculum" / "install.core.json").is_file():
            return candidate
    return cur


def manifest_path(root: Path, env: dict[str, str]) -> Path:
    override = env.get("UMBRACULUM_INSTALL_MANIFEST", "").strip()
    if override:
        return Path(override).resolve()

    active = root / ".umbraculum" / "install.json"
    if active.is_file():
        return active

    profile = env.get("UMBRACULUM_MODULE_PROFILE", "platform").strip() or "platform"
    name = "install.reference.json" if profile == "reference" else "install.core.json"
    return root / ".umbraculum" / name


def load_manifest(root: Path, env: dict[str, str]) -> dict:
    path = manifest_path(root, env)
    if not path.is_file():
        profile = env.get("UMBRACULUM_MODULE_PROFILE", "platform").strip() or "platform"
        if profile == "reference":
            return {
                "id": "reference",
                "verticals": ["brewery"],
                "canonical": ["automation", "pim", "mrp", "crp"],
                "nativeApps": ["brewery"],
            }
        return {
            "id": "core",
            "verticals": [],
            "canonical": ["automation", "pim", "mrp", "crp"],
            "nativeApps": ["blank"],
        }
    return json.loads(path.read_text(encoding="utf-8"))


def main() -> int:
    parser = argparse.ArgumentParser(description="Resolve Umbraculum installation profile")
    parser.add_argument("--repo-root", type=Path, default=None)
    parser.add_argument(
        "--field",
        choices=["id", "verticals", "canonical", "nativeApps", "primaryNativeApp", "hasBrewery"],
        default=None,
    )
    parser.add_argument("--json", action="store_true", help="Print full manifest JSON")
    args = parser.parse_args()

    root = args.repo_root.resolve() if args.repo_root else repo_root()
    env = dict(os.environ)
    manifest = load_manifest(root, env)

    if args.json:
        print(json.dumps(manifest, indent=2))
        return 0

    if args.field == "id":
        print(manifest.get("id", "core"))
    elif args.field == "verticals":
        print(",".join(manifest.get("verticals", [])))
    elif args.field == "canonical":
        print(",".join(manifest.get("canonical", [])))
    elif args.field == "nativeApps":
        print(",".join(manifest.get("nativeApps", [])))
    elif args.field == "primaryNativeApp":
        apps = manifest.get("nativeApps", ["blank"])
        print(apps[0] if apps else "blank")
    elif args.field == "hasBrewery":
        print("true" if "brewery" in manifest.get("verticals", []) else "false")
    else:
        print(json.dumps(manifest))

    return 0


if __name__ == "__main__":
    sys.exit(main())
